Library/Homebrew/dev-cmd/update-python-resources.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+require "cli/parser"
+
+module Homebrew
+  module_function
+
+  def update_python_resources_args
+    Homebrew::CLI::Parser.new do
+      usage_banner <<~EOS
+        `update-python-resources` [<options>] <formula>
+
+        Update versions for PyPI resource blocks in <formula>.
+      EOS
+      switch "-p", "--print-only",
+             description: "Print the updated resource blocks instead of changing <formula>."
+      min_named :formula
+    end
+  end
+
+  def update_python_resources
+    update_python_resources_args.parse
+
+    args.formulae.each do |formula|
+      pypi_name = if formula.stable.url.start_with?("https://files.pythonhosted.org/packages/")
+        File.basename(formula.stable.url).match(/^(.+)-[a-z0-9.]+$/)[1]
+      else
+        formula.name
+      end
+
+      pypi_resources = formula.resources.select do |resource|
+        resource.url.start_with?("https://files.pythonhosted.org/packages/") && resource.name != "homebrew-virtualenv"
+      end
+
+      odie "No PyPI resources found in \"#{formula.name}\"" if pypi_resources.empty?
+
+      unless Formula["python@3.8"].any_version_installed?
+        odie '"python@3.8" must be installed (`brew install python@3.8`)'
+      end
+
+      odie '"pipgrip" must be installed (`brew install pipgrip`)' unless Formula["pipgrip"].any_version_installed?
+
+      ohai "Retrieving PyPI dependencies for \"#{pypi_name}==#{formula.version}\"" unless args.print_only?
+      pipgrip_output = Utils.popen_read Formula["pipgrip"].bin/"pipgrip", "--json", "--no-cache-dir",
+                                        "#{pypi_name}==#{formula.version}"
+      unable_to_resolve_deps_error formula.name unless $CHILD_STATUS.success?
+
+      dependencies = JSON.parse(pipgrip_output.downcase)
+
+      ohai "Updating `resource` blocks"
```

Comment:
Small nit - can the backticks be removed since "resource" isn't a command?

Reply:
Sure. I added them, decided I didn't love them, and then forgot to remove them... 😄